classes/sota.bbclass

```diff
@@ -1,14 +1,14 @@
 DISTROOVERRIDES .= "${@bb.utils.contains('DISTRO_FEATURES', 'sota', ':sota', '', d)}"
 
-HOSTTOOLS_NONFATAL += "java"
+HOSTTOOLS_NONFATAL += "java repo python"
 
 SOTA_CLIENT ??= "aktualizr"
 SOTA_CLIENT_PROV ??= "aktualizr-shared-prov"
 SOTA_DEPLOY_CREDENTIALS ?= "1"
 SOTA_HARDWARE_ID ??= "${MACHINE}"
 
 IMAGE_INSTALL_append_sota = " ostree os-release ${SOTA_CLIENT} ${SOTA_CLIENT_PROV}"
-IMAGE_CLASSES += " image_types_ostree image_types_ota"
+IMAGE_CLASSES += " image_types_ostree image_types_ota image_repo_manifest"
```

Comment:
Thanks for the submission! What is the benefit of adding to `IMAGE_CLASSES` instead of inheriting? I couldn't find guidance in the docs.

And why is it better to add to `HOSTTOOLS_NONFATAL` here instead of in `image_repo_manifest.bbclass`?

In general, looks like it can't hurt, but if we do go through with this, we need to fix the comment in `image_repo_manifest.bbclass` that mentions inheriting.

Reply:
Hi, Patrick:

Thanks for reviewing my patch.

Adding it to IMAGE_CLASSES would only impact image recipes, but inheriting it in sota.bbclass would impact all recipes, like it will end up with adding a IMAGE_PREPROCESS_COMMAND command for aktualizr, ostree..., which makes no sense.

Putting HOSTTOOLS_NONFATAL into image_repo_manifest.bbclass was not a problem, but usually HOSTTOOLS_NONFATAL is a global variable (set in bitbake.conf, base.bbclass in OE), and we already have a HOSTTOOLS_NONFATAL setting in sota.bbclass, so my intention was to combine them to avoiding being redundant.

Reply:
> Adding it to IMAGE_CLASSES would only impact image recipes, but inheriting it in sota.bbclass would impact all recipes, like it will end up with adding a IMAGE_PREPROCESS_COMMAND command for aktualizr, ostree..., which makes no sense

Okay, that makes sense to me now. Thanks for clearing that up!

> Putting HOSTTOOLS_NONFATAL into image_repo_manifest.bbclass was not a problem, but usually HOSTTOOLS_NONFATAL is a global variable (set in bitbake.conf, base.bbclass in OE), and we already have a HOSTTOOLS_NONFATAL setting in sota.bbclass, so my intention was to combine them to avoiding being redundant.

This part I'm less sure of. If anyone were to reuse `image_repo_manifest.bbclass`, it would then be less obvious that `repo` and `python` are (optional) dependencies.